### ISSUE TITLE
How to get each sample loss of batch

### ISSUE THREAD

**Issue by gaurav-kjain** [fchollet/keras#3984]
Is it possible to log the each batch sample loss during batch processing and get the loss list in on_batch_end(), as it returns total loss?

something like

[sample id1-loss_val, sample],[sample-id2,loss-val2]......

**Comment by AdityaGudimella** [comment]
You can do that using the callbacks by writing your own as given in the documentation.

<code>
class LossHistory(keras.callbacks.Callback):
    
    def on_train_begin(self, logs={}):
        self.losses = []

    def on_batch_end(self, batch, logs={}):
        self.losses.append(logs.get('loss'))
</code>

**Comment by mgraczyk** [comment]
@AdityaGudimella It looks like your snippet gives the batch loss, not the sample loss, right?

**Comment by Farbdose** [comment]
@gaurav-kjain @AdityaGudimella ping
Any solutions for this problem?

**Comment by jbkoh** [comment]
I would like to upvote this issue as I (as well as some others) need such information for active learning setup.

Thanks!
